packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts

```diff
@@ -164,6 +182,20 @@ export class GLSPDiagramWidget extends DiagramWidget implements SaveableSource {
         }
         return undefined;
     }
+
+    protected async getSelectedElementIds(): Promise<string[]> {
+        const selectedElementIds = await this.actionDispatcher.request(GetSelectionAction.create())
```

Comment:
You could use the `EditorContextService` here to get the current selection instead of reexporting the sprotty `GetSelectionAction` in the client.

```typescript
    protected async getSelectedElementIds(): Promise<string[]> {
        const editorContextService= this.diContainer.get(EditorContextService);
        return editorContextService.selectedElements.map(element=>element.id);
    }
```

Reply:
Thanks for that hint, I incorporated that snipped and it works great!